apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -117,23 +104,42 @@ defmodule GameBackend.Matches do
   ####################
   #      Helpers     #
   ####################
+  defp match_prestige_reward(nil, position, rewards) do
+    match_prestige_reward(%{amount: 0}, position, rewards)
+  end
 
-  ## TODO: Properly pre-process `currencies_config` so the keys are integers and we don't need conversion
-  ##    https://github.com/lambdaclass/mirra_backend/issues/601
-  def get_amount_of_trophies_to_modify(current_trophies, position, currencies_config) when is_integer(position) do
-    get_amount_of_trophies_to_modify(current_trophies, to_string(position), currencies_config)
+  defp match_prestige_reward(prestige, position, rewards) do
+    Map.get(rewards, position)
+    |> Enum.find(fn %{min: minp, max: maxp} -> prestige.amount in minp..maxp end)
+    |> Map.get(:reward)
   end
 
-  def get_amount_of_trophies_to_modify(current_trophies, position, currencies_config) do
-    Enum.sort_by(
-      get_in(currencies_config, ["ranking_system", "ranks"]),
-      fn %{"maximum_rank" => maximum} -> maximum end,
-      :asc
-    )
-    |> Enum.find(get_in(currencies_config, ["ranking_system", "infinite_rank"]), fn %{"maximum_rank" => maximum} ->
-      maximum > current_trophies
-    end)
-    |> Map.get(position)
+  defp calculate_rank_and_amount_changes(nil, reward, ranks) do
+    calculate_rank_and_amount_changes(%{amount: 0}, reward, ranks)
+  end
+
+  defp calculate_rank_and_amount_changes(prestige, reward, ranks) when reward >= 0 do
+    amount = prestige.amount + reward
+    new_rank = Enum.find(ranks, fn rank -> amount in rank.min..rank.max end)
+    %{rank: new_rank.rank, sub_rank: new_rank.sub_rank, amount: amount}
+  end
+
+  defp calculate_rank_and_amount_changes(prestige, reward, ranks) when reward < 0 do
+    current_rank = Enum.find(ranks, fn rank -> prestige.amount in rank.min..rank.max end)
+    amount = max(prestige.amount + reward, current_rank.min)
+    %{amount: amount}
```

Comment:
I think all this logic should be under CurseOfMirra's path. It's business for CurseOfMirra only.